### PR TITLE
Added pause and win menus

### PR DESCRIPTION
**Pull request by jachym-stanek**
Added menu GUIs to handle victory and game pause events

### CODE REVIEW

File: src/pause.c [hunk 1]
```diff
@@ -1,13 +1,204 @@
...
+  	if (parled_mem_base == NULL) {
+    	return false;
+  	}
+
+  	if (!knobs_init()){
+  		return false;
+  	}
+
+	parlcd_hx8357_init(parlcd_mem_base);
+ 	parlcd_write_cmd(parlcd_mem_base, 0x2c);
```
Comment (Hebcak) [inline src/pause.c:42]: It is uneccessary to reinitialize the display

**Comment by Hebcak** [comment]
And please always add the new files to the Makefile

File: src/pause.h [hunk 2]
```diff
@@ -1,5 +1,34 @@
 //header file for pause.c
 
+#include <stdbool.h>
+
+#include "game.h"
```
Comment (Hebcak) [inline src/pause.h:5]: Unnecessary import

File: src/pause.h [hunk 3]
```diff
@@ -1,5 +1,34 @@
...
+
+void info_pause(void);
+
+void light_leds_pause(void);
+
+void tab_pressed_pause(void);
+
 //operations to be done on pause
-void pause_oper();
+void pause_oper(game_struct * game);
```
Comment (Hebcak) [inline src/pause.h:29]: unnecessary argument

**Comment by Hebcak** [comment]
please make all files use the same `display_data(uint16_t * data, unsigned char * parlcd_mem_base)` function. Including main_menu.c

**Comment by Hebcak** [comment]
also please use newest build when making changes

**Comment by Hebcak** [comment]
make it stop printing the player names so often

File: src/pause.c [hunk 4]
```diff
@@ -1,13 +1,204 @@
...
+	while (!PAUSE_DATA.end){
+		// input from keyboard
+		t = getc_timeout(STDIN_FILENO, READ_TIMEOUT, &in);
+
+		if (t == 1){
+			switch (in) {
+				case '1':
+				case '2':
+					PAUSE_DATA.tab = in;	// tell probram new position to fill
+					menu_graphics();
```
Comment (Hebcak) [inline src/pause.c:162]: Change to pause graphics (and in other places too)

File: src/pause.c [hunk 5]
```diff
@@ -1,13 +1,204 @@
...
 
 //operations to be done on pause
-void pause_oper() {
+void pause_oper(game_struct * game){
 	printf("The game has been paused!\n");
-	sleep(1);
-	while (!get_pause()) {}
+	init_perifs_pause();
+	pause_graphics();
+	pause_loop();
```
Comment (Hebcak) [inline src/pause.c:130]: add here `PAUSE_DATA.end = false;`

**Comment by jachym-stanek** [comment]
Notes taken, new pull should fix the issues

**Pull request closed without merging**